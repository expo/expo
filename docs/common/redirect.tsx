import Head from 'next/head';

const redirect = (destination: string) => {
  return () => (
    <Head>
      <meta httpEquiv="refresh" content={`0; url=${destination}`} />
    </Head>
  );
};

export default redirect;
