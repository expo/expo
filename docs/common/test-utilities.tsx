import { TooltipProvider } from '@radix-ui/react-tooltip';
import { render, RenderOptions } from '@testing-library/react';
import GithubSlugger from 'github-slugger';
import { RouterContext } from 'next/dist/shared/lib/router-context.shared-runtime';
import { type NextRouter } from 'next/router';
import { type PropsWithChildren, type ReactElement } from 'react';

import { HeadingManager } from '~/common/headingManager';
import { HeadingsContext } from '~/components/page-higher-order/withHeadingManager';

const Wrapper = ({ children }: PropsWithChildren) => (
  <TooltipProvider>
    <HeadingsContext.Provider value={new HeadingManager(new GithubSlugger(), { headings: [] })}>
      {children}
    </HeadingsContext.Provider>
  </TooltipProvider>
);

export function renderWithHeadings(
  element: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) {
  return render(element, { wrapper: Wrapper, ...options });
}

export function renderWithTestRouter(element: ReactElement, router: Partial<NextRouter> = {}) {
  return render(
    <RouterContext.Provider value={router as NextRouter}>{element}</RouterContext.Provider>
  );
}
