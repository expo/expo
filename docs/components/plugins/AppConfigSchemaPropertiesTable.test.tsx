import { screen } from '@testing-library/react';

import AppConfigSchemaPropertiesTable, {
  formatSchema,
  createDescription,
  Property,
} from './AppConfigSchemaPropertiesTable';

import { renderWithHeadings } from '~/common/test-utilities';

const testSchema: Record<string, Property> = {
  name: {
    description: 'Name of your app.',
    type: 'string',
    meta: {
      bareWorkflow: "Edit the 'Display Name' field in Xcode",
    },
  },
  androidNavigationBar: {
    description: 'Configuration for the bottom navigation bar on Android.',
    type: 'object',
    properties: {
      visible: {
        description: 'Determines how and when the navigation bar is shown.',
        type: 'string',
        properties: {
          always: {
            description: 'Test sub-sub-property',
            type: 'boolean',
          },
        },
        meta: {
          expoKit: 'Set this property using Xcode.',
        },
        enum: ['leanback', 'immersive', 'sticky-immersive'],
      },
      backgroundColor: {
        description: 'Specifies the background color of the navigation bar. ',
        type: 'string',
        pattern: '^#|(&#x23;)\\d{6}$',
        meta: {
          regexHuman: "6 character long hex color string, eg: `'#000000'`",
        },
      },
    },
    meta: {
      expoKit: 'Set this property using AppConstants.java.',
      bareWorkflow: 'Set this property using just Xcode',
    },
  },
  intentFilters: {
    description: 'Configuration for setting an array of custom intent filters in Android manifest.',
    example: [
      {
        autoVerify: true,
        data: {
          host: '*.example.com',
        },
      },
    ],
    exampleString: '\n [{ \n "autoVerify": true, \n "data": {"host": "*.example.com" \n } \n }]',
    type: 'array',
    uniqueItems: true,
    items: {
      type: 'object',
      properties: {
        autoVerify: {
          description:
            'You may also use an intent filter to set your app as the default handler for links',
          type: 'boolean',
        },
        data: {
          type: ['array', 'object'],
          items: {
            type: 'object',
            properties: {
              host: { description: 'the host, e.g. `example.com`', type: 'string' },
            },
            additionalProperties: false,
          },
          properties: {
            host: { type: 'string' },
          },
          additionalProperties: false,
        },
      },
      additionalProperties: false,
      required: ['action'],
    },
    meta: {
      bareWorkflow: 'This is set in AndroidManifest.xml directly.',
    },
  },
};

describe('AppConfigSchemaPropertiesTable', () => {
  test('correctly matches snapshot', () => {
    const { container } = renderWithHeadings(
      <AppConfigSchemaPropertiesTable schema={testSchema} />
    );
    expect(container).toMatchSnapshot();
  });

  test('description includes all required components', () => {
    renderWithHeadings(
      <AppConfigSchemaPropertiesTable schema={{ entry: testSchema.androidNavigationBar }} />
    );

    expect(screen.getByText('Specifies the background color of the navigation bar.'));
    expect(screen.getByText('6 character long hex color string, eg:'));
    expect(screen.getByText('Set this property using just Xcode'));
    expect(screen.getByText('Set this property using AppConstants.java.'));
  });
});

describe('formatSchema', () => {
  const formattedSchema = formatSchema(Object.entries(testSchema));
  test('name is property at root level', () => {
    expect(formattedSchema[0].name).toBe('name');
  });
  test('androidNavigationBar has two subproperties', () => {
    expect(formattedSchema[1].subproperties.length).toBe(2);
  });
  test('visible is androidNavigationBar subproperty', () => {
    expect(formattedSchema[1].subproperties[0].name).toBe('visible');
  });
  test('always is visible subproperty', () => {
    expect(formattedSchema[1].subproperties[0].subproperties[0].name).toBe('always');
  });
  test('backgroundColor is androidNavigationBar subproperty', () => {
    expect(formattedSchema[1].subproperties[1].name).toBe('backgroundColor');
  });
  test('intentFilters is property at root level', () => {
    expect(formattedSchema[2].name).toBe('intentFilters');
  });
  test('autoVerify is intentFilters subproperty', () => {
    expect(formattedSchema[2].subproperties[0].name).toBe('autoVerify');
  });
  test('data is intentFilters subproperty', () => {
    expect(formattedSchema[2].subproperties[1].name).toBe('data');
  });
  test('host is data subproperty', () => {
    expect(formattedSchema[2].subproperties[1].subproperties[0].name).toBe('host');
  });
});

describe('createDescription', () => {
  test('type and description are rendered correctly', () => {
    const intentFiltersObject = Object.entries(testSchema)[2];
    const intentFiltersObjectValue = intentFiltersObject[1] as any;
    const result = createDescription(intentFiltersObject);

    expect(result).toBe(`${intentFiltersObjectValue.description}`);
  });

  test('regexHuman is added correctly', () => {
    //Note: to access this subproperty is tedious without a call to formatSchema
    const backgroundColorObject = Object.entries(Object.values(testSchema)[1].properties!)[1];
    const backgroundColorObjectValue = backgroundColorObject[1];
    const result = createDescription(backgroundColorObject);

    expect(result).toBe(
      `${backgroundColorObjectValue.description}\n\n${backgroundColorObjectValue.meta!.regexHuman}`
    );
  });
});
