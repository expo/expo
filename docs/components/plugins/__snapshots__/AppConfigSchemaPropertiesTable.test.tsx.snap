// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppConfigSchemaPropertiesTable correctly matches snapshot 1`] = `
<div>
  <div
    class="!pb-4 last:[&>*]:!mb-1 css-c2t1qw-AppConfigProperty"
  >
    <p
      class="  css-1kfqm4n-TextComponent"
      data-text="true"
    >
      <a
        class="css-1yk6eqh"
        href="#name"
      >
        <span
          class="css-s3qkfm"
          id="name"
        />
        <span
          class="css-6n7j50"
        >
          <code
            class="css-2z1nlz-PropertyName"
            data-text="true"
          >
            name
          </code>
        </span>
        <span
          class="css-13330di"
        >
          <svg
            aria-label="permalink"
            class="anchor-icon"
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
              stroke="#9B9B9B"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
            <path
              d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
              stroke="#9B9B9B"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </span>
      </a>
    </p>
    <p
      class="css-1lwc34i-AppConfigProperty"
      data-text="true"
    >
      Type: 
      <code
        class="css-iwb7ur-TextComponent"
        data-text="true"
      >
        string
      </code>
    </p>
    <p
      class="mb-4 css-1kfqm4n-TextComponent"
      data-text="true"
    >
      Name of your app.
    </p>
    <details
      class="css-xpa10j"
      id="bare-workflow"
    >
      <summary
        class="css-zjlxe1"
      >
        <div
          class="css-qos8r6"
        >
          <svg
            class="icon-sm text-icon-default css-nlesit"
            fill="none"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="triangle-down"
            >
              <path
                d="M18.8831 9.43005H5.11694L12 16.3131L18.8831 9.43005Z"
                fill="currentColor"
                id="Vector 250"
              />
            </g>
          </svg>
        </div>
        <a
          href="#bare-workflow"
        >
          <span
            class="css-1n5kx9e-TextComponent"
            data-text="true"
          >
            Bare Workflow
          </span>
          <span
            class="css-13330di"
          >
            <svg
              aria-label="permalink"
              class="anchor-icon"
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
                stroke="#9B9B9B"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
                stroke="#9B9B9B"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </span>
        </a>
      </summary>
      <div
        class="last:[&>*]:!mb-1 css-qbaxbl"
      >
        <p
          class="mb-4 css-1kfqm4n-TextComponent"
          data-text="true"
        >
          Edit the 'Display Name' field in Xcode
        </p>
      </div>
    </details>
    <div />
  </div>
  <div
    class="!pb-4 last:[&>*]:!mb-1 css-c2t1qw-AppConfigProperty"
  >
    <p
      class="  css-1kfqm4n-TextComponent"
      data-text="true"
    >
      <a
        class="css-1yk6eqh"
        href="#androidnavigationbar"
      >
        <span
          class="css-s3qkfm"
          id="androidnavigationbar"
        />
        <span
          class="css-6n7j50"
        >
          <code
            class="css-2z1nlz-PropertyName"
            data-text="true"
          >
            androidNavigationBar
          </code>
        </span>
        <span
          class="css-13330di"
        >
          <svg
            aria-label="permalink"
            class="anchor-icon"
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
              stroke="#9B9B9B"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
            <path
              d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
              stroke="#9B9B9B"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </span>
      </a>
    </p>
    <p
      class="css-1lwc34i-AppConfigProperty"
      data-text="true"
    >
      Type: 
      <code
        class="css-iwb7ur-TextComponent"
        data-text="true"
      >
        object
      </code>
    </p>
    <p
      class="mb-4 css-1kfqm4n-TextComponent"
      data-text="true"
    >
      Configuration for the bottom navigation bar on Android.
    </p>
    <details
      class="css-xpa10j"
      id="expokit"
    >
      <summary
        class="css-zjlxe1"
      >
        <div
          class="css-qos8r6"
        >
          <svg
            class="icon-sm text-icon-default css-nlesit"
            fill="none"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="triangle-down"
            >
              <path
                d="M18.8831 9.43005H5.11694L12 16.3131L18.8831 9.43005Z"
                fill="currentColor"
                id="Vector 250"
              />
            </g>
          </svg>
        </div>
        <a
          href="#expokit"
        >
          <span
            class="css-1n5kx9e-TextComponent"
            data-text="true"
          >
            ExpoKit
          </span>
          <span
            class="css-13330di"
          >
            <svg
              aria-label="permalink"
              class="anchor-icon"
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
                stroke="#9B9B9B"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
                stroke="#9B9B9B"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </span>
        </a>
      </summary>
      <div
        class="last:[&>*]:!mb-1 css-qbaxbl"
      >
        <p
          class="mb-4 css-1kfqm4n-TextComponent"
          data-text="true"
        >
          Set this property using AppConstants.java.
        </p>
      </div>
    </details>
    <details
      class="css-xpa10j"
      id="bare-workflow-1"
    >
      <summary
        class="css-zjlxe1"
      >
        <div
          class="css-qos8r6"
        >
          <svg
            class="icon-sm text-icon-default css-nlesit"
            fill="none"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="triangle-down"
            >
              <path
                d="M18.8831 9.43005H5.11694L12 16.3131L18.8831 9.43005Z"
                fill="currentColor"
                id="Vector 250"
              />
            </g>
          </svg>
        </div>
        <a
          href="#bare-workflow-1"
        >
          <span
            class="css-1n5kx9e-TextComponent"
            data-text="true"
          >
            Bare Workflow
          </span>
          <span
            class="css-13330di"
          >
            <svg
              aria-label="permalink"
              class="anchor-icon"
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
                stroke="#9B9B9B"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
                stroke="#9B9B9B"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </span>
        </a>
      </summary>
      <div
        class="last:[&>*]:!mb-1 css-qbaxbl"
      >
        <p
          class="mb-4 css-1kfqm4n-TextComponent"
          data-text="true"
        >
          Set this property using just Xcode
        </p>
      </div>
    </details>
    <div>
      <div
        class="!pb-4 last:[&>*]:!mb-1 css-c2t1qw-AppConfigProperty"
      >
        <p
          class="  css-1kfqm4n-TextComponent"
          data-text="true"
        >
          <a
            class="css-1yk6eqh"
            href="#visible"
          >
            <span
              class="css-s3qkfm"
              id="visible"
            />
            <span
              class="css-6n7j50"
            >
              <code
                class="css-2z1nlz-PropertyName"
                data-text="true"
              >
                visible
              </code>
            </span>
            <span
              class="css-13330di"
            >
              <svg
                aria-label="permalink"
                class="anchor-icon"
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
                  stroke="#9B9B9B"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
                <path
                  d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
                  stroke="#9B9B9B"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </span>
          </a>
        </p>
        <p
          class="css-1lwc34i-AppConfigProperty"
          data-text="true"
        >
          Type: 
          <code
            class="css-iwb7ur-TextComponent"
            data-text="true"
          >
            enum
          </code>
           • Path:
           
          <code
            class="css-te23k8-AppConfigProperty"
          >
            androidNavigationBar
            .
            visible
          </code>
        </p>
        <p
          class="mb-4 css-1kfqm4n-TextComponent"
          data-text="true"
        >
          Determines how and when the navigation bar is shown.
        </p>
        <details
          class="css-xpa10j"
          id="expokit-1"
        >
          <summary
            class="css-zjlxe1"
          >
            <div
              class="css-qos8r6"
            >
              <svg
                class="icon-sm text-icon-default css-nlesit"
                fill="none"
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  id="triangle-down"
                >
                  <path
                    d="M18.8831 9.43005H5.11694L12 16.3131L18.8831 9.43005Z"
                    fill="currentColor"
                    id="Vector 250"
                  />
                </g>
              </svg>
            </div>
            <a
              href="#expokit-1"
            >
              <span
                class="css-1n5kx9e-TextComponent"
                data-text="true"
              >
                ExpoKit
              </span>
              <span
                class="css-13330di"
              >
                <svg
                  aria-label="permalink"
                  class="anchor-icon"
                  fill="none"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
                    stroke="#9B9B9B"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                  <path
                    d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
                    stroke="#9B9B9B"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </span>
            </a>
          </summary>
          <div
            class="last:[&>*]:!mb-1 css-qbaxbl"
          >
            <p
              class="mb-4 css-1kfqm4n-TextComponent"
              data-text="true"
            >
              Set this property using Xcode.
            </p>
          </div>
        </details>
        <div>
          <div
            class="!pb-4 last:[&>*]:!mb-1 css-c2t1qw-AppConfigProperty"
          >
            <p
              class="  css-1kfqm4n-TextComponent"
              data-text="true"
            >
              <a
                class="css-1yk6eqh"
                href="#always"
              >
                <span
                  class="css-s3qkfm"
                  id="always"
                />
                <span
                  class="css-6n7j50"
                >
                  <code
                    class="css-2z1nlz-PropertyName"
                    data-text="true"
                  >
                    always
                  </code>
                </span>
                <span
                  class="css-13330di"
                >
                  <svg
                    aria-label="permalink"
                    class="anchor-icon"
                    fill="none"
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
                      stroke="#9B9B9B"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                    <path
                      d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
                      stroke="#9B9B9B"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </span>
              </a>
            </p>
            <p
              class="css-1lwc34i-AppConfigProperty"
              data-text="true"
            >
              Type: 
              <code
                class="css-iwb7ur-TextComponent"
                data-text="true"
              >
                boolean
              </code>
               • Path:
               
              <code
                class="css-te23k8-AppConfigProperty"
              >
                androidNavigationBar.visible
                .
                always
              </code>
            </p>
            <p
              class="mb-4 css-1kfqm4n-TextComponent"
              data-text="true"
            >
              Test sub-sub-property
            </p>
            <div />
          </div>
        </div>
      </div>
      <div
        class="!pb-4 last:[&>*]:!mb-1 css-c2t1qw-AppConfigProperty"
      >
        <p
          class="  css-1kfqm4n-TextComponent"
          data-text="true"
        >
          <a
            class="css-1yk6eqh"
            href="#backgroundcolor"
          >
            <span
              class="css-s3qkfm"
              id="backgroundcolor"
            />
            <span
              class="css-6n7j50"
            >
              <code
                class="css-2z1nlz-PropertyName"
                data-text="true"
              >
                backgroundColor
              </code>
            </span>
            <span
              class="css-13330di"
            >
              <svg
                aria-label="permalink"
                class="anchor-icon"
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
                  stroke="#9B9B9B"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
                <path
                  d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
                  stroke="#9B9B9B"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </span>
          </a>
        </p>
        <p
          class="css-1lwc34i-AppConfigProperty"
          data-text="true"
        >
          Type: 
          <code
            class="css-iwb7ur-TextComponent"
            data-text="true"
          >
            string
          </code>
           • Path:
           
          <code
            class="css-te23k8-AppConfigProperty"
          >
            androidNavigationBar
            .
            backgroundColor
          </code>
        </p>
        <p
          class="mb-4 css-1kfqm4n-TextComponent"
          data-text="true"
        >
          Specifies the background color of the navigation bar.
        </p>
        

        <p
          class="mb-4 css-1kfqm4n-TextComponent"
          data-text="true"
        >
          6 character long hex color string, eg: 
          <code
            class="css-1a6y07t-TextComponent"
            data-text="true"
          >
            '#000000'
          </code>
        </p>
        <div />
      </div>
    </div>
  </div>
  <div
    class="!pb-4 last:[&>*]:!mb-1 css-c2t1qw-AppConfigProperty"
  >
    <p
      class="  css-1kfqm4n-TextComponent"
      data-text="true"
    >
      <a
        class="css-1yk6eqh"
        href="#intentfilters"
      >
        <span
          class="css-s3qkfm"
          id="intentfilters"
        />
        <span
          class="css-6n7j50"
        >
          <code
            class="css-2z1nlz-PropertyName"
            data-text="true"
          >
            intentFilters
          </code>
        </span>
        <span
          class="css-13330di"
        >
          <svg
            aria-label="permalink"
            class="anchor-icon"
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
              stroke="#9B9B9B"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
            <path
              d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
              stroke="#9B9B9B"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </span>
      </a>
    </p>
    <p
      class="css-1lwc34i-AppConfigProperty"
      data-text="true"
    >
      Type: 
      <code
        class="css-iwb7ur-TextComponent"
        data-text="true"
      >
        array
      </code>
    </p>
    <p
      class="mb-4 css-1kfqm4n-TextComponent"
      data-text="true"
    >
      Configuration for setting an array of custom intent filters in Android manifest.
    </p>
    <details
      class="css-xpa10j"
      id="bare-workflow-2"
    >
      <summary
        class="css-zjlxe1"
      >
        <div
          class="css-qos8r6"
        >
          <svg
            class="icon-sm text-icon-default css-nlesit"
            fill="none"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="triangle-down"
            >
              <path
                d="M18.8831 9.43005H5.11694L12 16.3131L18.8831 9.43005Z"
                fill="currentColor"
                id="Vector 250"
              />
            </g>
          </svg>
        </div>
        <a
          href="#bare-workflow-2"
        >
          <span
            class="css-1n5kx9e-TextComponent"
            data-text="true"
          >
            Bare Workflow
          </span>
          <span
            class="css-13330di"
          >
            <svg
              aria-label="permalink"
              class="anchor-icon"
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
                stroke="#9B9B9B"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
                stroke="#9B9B9B"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </span>
        </a>
      </summary>
      <div
        class="last:[&>*]:!mb-1 css-qbaxbl"
      >
        <p
          class="mb-4 css-1kfqm4n-TextComponent"
          data-text="true"
        >
          This is set in AndroidManifest.xml directly.
        </p>
      </div>
    </details>
    <blockquote
      class="css-zknray-Callout"
      data-testid="callout-container"
    >
      <div
        class="css-7jywx8-Callout"
      >
        <svg
          class="icon-sm text-icon-default"
          fill="currentColor"
          role="img"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="info-circle"
          >
            <path
              clip-rule="evenodd"
              d="M12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1ZM12 7C11.4477 7 11 7.44772 11 8C11 8.55228 11.4477 9 12 9H12.01C12.5623 9 13.01 8.55228 13.01 8C13.01 7.44772 12.5623 7 12.01 7H12ZM13 12C13 11.4477 12.5523 11 12 11C11.4477 11 11 11.4477 11 12V16C11 16.5523 11.4477 17 12 17C12.5523 17 13 16.5523 13 16V12Z"
              fill-rule="evenodd"
              id="Solid"
            />
          </g>
        </svg>
      </div>
      <div
        class="css-1m74yfi-Callout"
      >
        <strong
          class="css-bvflvn-TextComponent"
        >
          Example
        </strong>
        <p
          class="mb-4 css-1kfqm4n-TextComponent"
          data-text="true"
        >
          [{  "autoVerify": true,  "data": {"host": "*.example.com"  }  }]
        </p>
      </div>
    </blockquote>
    <div>
      <div
        class="!pb-4 last:[&>*]:!mb-1 css-c2t1qw-AppConfigProperty"
      >
        <p
          class="  css-1kfqm4n-TextComponent"
          data-text="true"
        >
          <a
            class="css-1yk6eqh"
            href="#autoverify"
          >
            <span
              class="css-s3qkfm"
              id="autoverify"
            />
            <span
              class="css-6n7j50"
            >
              <code
                class="css-2z1nlz-PropertyName"
                data-text="true"
              >
                autoVerify
              </code>
            </span>
            <span
              class="css-13330di"
            >
              <svg
                aria-label="permalink"
                class="anchor-icon"
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
                  stroke="#9B9B9B"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
                <path
                  d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
                  stroke="#9B9B9B"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </span>
          </a>
        </p>
        <p
          class="css-1lwc34i-AppConfigProperty"
          data-text="true"
        >
          Type: 
          <code
            class="css-iwb7ur-TextComponent"
            data-text="true"
          >
            boolean
          </code>
           • Path:
           
          <code
            class="css-te23k8-AppConfigProperty"
          >
            intentFilters
            .
            autoVerify
          </code>
        </p>
        <p
          class="mb-4 css-1kfqm4n-TextComponent"
          data-text="true"
        >
          You may also use an intent filter to set your app as the default handler for links
        </p>
        <div />
      </div>
      <div
        class="!pb-4 last:[&>*]:!mb-1 css-c2t1qw-AppConfigProperty"
      >
        <p
          class="  css-1kfqm4n-TextComponent"
          data-text="true"
        >
          <a
            class="css-1yk6eqh"
            href="#data"
          >
            <span
              class="css-s3qkfm"
              id="data"
            />
            <span
              class="css-6n7j50"
            >
              <code
                class="css-2z1nlz-PropertyName"
                data-text="true"
              >
                data
              </code>
            </span>
            <span
              class="css-13330di"
            >
              <svg
                aria-label="permalink"
                class="anchor-icon"
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
                  stroke="#9B9B9B"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
                <path
                  d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
                  stroke="#9B9B9B"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </span>
          </a>
        </p>
        <p
          class="css-1lwc34i-AppConfigProperty"
          data-text="true"
        >
          Type: 
          <code
            class="css-iwb7ur-TextComponent"
            data-text="true"
          >
            array || object
          </code>
           • Path:
           
          <code
            class="css-te23k8-AppConfigProperty"
          >
            intentFilters
            .
            data
          </code>
        </p>
        <div>
          <div
            class="!pb-4 last:[&>*]:!mb-1 css-c2t1qw-AppConfigProperty"
          >
            <p
              class="  css-1kfqm4n-TextComponent"
              data-text="true"
            >
              <a
                class="css-1yk6eqh"
                href="#host"
              >
                <span
                  class="css-s3qkfm"
                  id="host"
                />
                <span
                  class="css-6n7j50"
                >
                  <code
                    class="css-2z1nlz-PropertyName"
                    data-text="true"
                  >
                    host
                  </code>
                </span>
                <span
                  class="css-13330di"
                >
                  <svg
                    aria-label="permalink"
                    class="anchor-icon"
                    fill="none"
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M10 13a5 5 0 007.54.54l3-3a5 5 0 00-7.07-7.07l-1.72 1.71"
                      stroke="#9B9B9B"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                    <path
                      d="M14 11a5 5 0 00-7.54-.54l-3 3a5 5 0 007.07 7.07l1.71-1.71"
                      stroke="#9B9B9B"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </span>
              </a>
            </p>
            <p
              class="css-1lwc34i-AppConfigProperty"
              data-text="true"
            >
              Type: 
              <code
                class="css-iwb7ur-TextComponent"
                data-text="true"
              >
                string
              </code>
               • Path:
               
              <code
                class="css-te23k8-AppConfigProperty"
              >
                intentFilters.data
                .
                host
              </code>
            </p>
            <div />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
